flake.nix

```diff
@@ -0,0 +1,199 @@
+{
+  description = "Decrypt and encrypt agenix secrets inside Emacs";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-22.11";
+
+    bash-strict-mode = {
+      url = "github:sellout/bash-strict-mode";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    homeManager = {
+      url = "github:nix-community/home-manager/release-22.11";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs:
+    {
+      overlays = {
+        default = final: prev: {
+          emacsPackagesFor = emacs:
+            (prev.emacsPackagesFor emacs).overrideScope'
+            (inputs.self.overlays.emacs final prev);
+        };
+
+        emacs = final: prev: efinal: eprev: {
+          agenix = inputs.self.packages.${final.system}.agenix-el;
+        };
```

Comment:
Is there a point to expose it separately? What's the usecase someone would use this instead of `overlays.default`?

Reply:
I provide something like this for any package that’s encapsulated in some “withPackages” like thing. In this case, it would be if someone was not using `emacsPackagesFor` for some reason. `emacsPackagesFor` is very general, so It might not be useful. In the Nix/Haskell world, you need to add the Haskell package explicitly to each compiler you support, so if the `default` overlay misses one, or a user adds their own, then the `haskell` overlay needs to be used directly to make it available via `withPackages`.

Reply:
Ohh I see, indeed it can be useful for someone